scrape/manager_test.go

```diff
@@ -610,6 +610,64 @@ func TestManagerTargetsUpdates(t *testing.T) {
 	}
 }
 
+func BenchmarkManagerReload(b *testing.B) {
+	opts := Options{}
+	testRegistry := prometheus.NewRegistry()
+	m, err := NewManager(&opts, nil, nil, testRegistry)
+	require.NoError(b, err)
+
+	m.scrapePools = map[string]*scrapePool{}
+	sp := &scrapePool{
+		activeTargets: map[uint64]*Target{},
+	}
+
+	for i := 0; i < b.N; i++ {
```

Comment:
It is essential to do the same amount of work each time the benchmark is run, so varying the number of targets with `b.N` is wrong.

Reply:
We are running the `reload` function 10k times, each time with 10k targets via `go test -bench=BenchmarkManagerReload -benchmem -run=- -count 6  -benchtime=10000x`. This is similar to this [benchmark](https://github.com/prometheus/prometheus/blob/main/tsdb/chunkenc/chunk_test.go#L266). 

Do you think it would be better to hardcode the #targets 10k instead?

Reply:
> This is similar to this [benchmark](https://github.com/prometheus/prometheus/blob/main/tsdb/chunkenc/chunk_test.go#L266).

Another invalid benchmark.

> Do you think it would be better to hardcode the #targets 10k instead?

Yes.